cmake/json_summary.cmake

```diff
@@ -0,0 +1,89 @@
+# ANSI codes
+unset(rst)
+unset(bld)
+# no color output on Windows or if disabled via CLICOLOR=0/CLICOLOR_FORCE=0
+if(NOT WIN32 AND NOT ("$ENV{CLICOLOR}" STREQUAL "0" OR "$ENV{CLICOLOR_FORCE}" STREQUAL "0"))
+    string(ASCII 27 esc)
+    set(rst "${esc}[0m") # reset
+    set(bld "${esc}[1m") # bold
+endif()
+
+#############################################################################
+# json_feature(
+#     var text
+#     [VALUES <value>...])
+#
+# Print feature info using <text> and the boolean value of <var> converted to
+# YES/NO.
+#
+# If additional values are given and <var> matches any of them, <var> is not
+# converted to YES/NO.
+#############################################################################
+
+function(json_feature var text)
+    cmake_parse_arguments(args "" "" "VALUES" ${ARGN})
+
+    set(state NO)
+    if(args_VALUES)
+        foreach(value ${args_VALUES})
+            if(${var} STREQUAL value)
+                set(state "${value}")
+                break()
+            endif()
+        endforeach()
+    elseif(${var})
+        set(state YES)
+    endif()
+
+    message("   ${text} ${bld}${state}${rst}")
+endfunction()
+
+#############################################################################
+# print feature summary
+#############################################################################
+
+message(STATUS "[nohmann_json]: Feature summary:")
+
+json_feature(JSON_BuildTests "Build tests?")
+if(JSON_BuildTests)
+    json_feature(JSON_32bitTest "Build the 32bit unit test?" VALUES AUTO ONLY)
+    json_feature(JSON_FastTests "Skip expensive/slow tests?")
+
+    if(JSON_TEST_DATA_DIRECTORY)
+        message("   Test data: ${JSON_TEST_DATA_DIRECTORY}")
+    else()
+        message("   Test data: ${JSON_TEST_DATA_URL} (v${JSON_TEST_DATA_VERSION})")
+    endif()
+
+    json_feature(JSON_Valgrind "Execute test suite with Valgrind?")
+    if(JSON_Valgrind)
+        message("   Valgrind command: ${CMAKE_MEMORYCHECK_COMMAND}" ${CMAKE_MEMORYCHECK_COMMAND_OPTIONS})
+    endif()
+
+    unset(test_cxx_standards)
```

Comment:
Prefer `set(var "")`. What if CMake learns to detect uses of undefined variables someday?